scripts/qa_crowbarsetup.sh

```diff
@@ -5130,7 +5130,7 @@ function onadmin_crowbar_nodeupgrade
             wait_for 300 5 "grep current_step $upgrade_progress_file | grep -v backup_openstack" "backup openstack step to finish"
 
             safely crowbarctl upgrade nodes all
-            wait_for 360 30 "crowbar_nodeupgrade_finished" "'nodes' upgrade step to finish"
+            wait_for 360 30 "crowbar_nodeupgrade_finished" "'nodes' upgrade step to finish" "complain 13 'Nodes step has failed. Check the upgrade status.'" "crowbarctl upgrade status nodes --plain |grep not_upgraded|cut -d ' ' -f 2 |cut -d '.' -f 1|sort|tr '\n' ','| sed 's/,$//'|xargs printf 'Upgrading nodes: %s\r'"
```

Comment:
1.
The value for `$error_cmd` is redundant. The `wait_for` function already prints a useful message. If its not useful then rather adapt the `waitfor` string. So my proposal is to pass an empty string here.

2.
The print while is a nice idea. If you add looong stuff as value though, it is a hint that you instead wanted a function and then rather just pass as function name as print_while string.

So I recommend to do this:

```
function show_crowbar_upgrade_status
{
    crowbarctl upgrade status nodes --plain | grep not_upgraded | cut -d ' ' -f 2 | cut -d '.' -f 1 | \
        sort | tr '\n' ',' | sed 's/,$//' | xargs printf 'Upgrading nodes: %s\r'
}

[...]

wait_for 360 30 "crowbar_nodeupgrade_finished" "'nodes' upgrade step to finish" "" "show_crowbar_upgrade_status"
```